docs/hello_nextflow/02_hello_channels.md

```diff
@@ -0,0 +1,868 @@
+
+### 2.4. Ensure the output file names will be unique
+
+We can continue publishing all the outputs to the same results directory, but we need to ensure they will have unique names.
+Specifically, we need to modify the first process to generate a file name dynamically so that the final file names will be unique.
+
+So how do we make the file names unique? A common way to do that is to use some unique piece of metadata as part of the file name.
+Here, for convenience, we'll just use the greeting itself since it's just a short string, and prepend it to the base output filename.
+
+In the process block, make the following code changes:
+
+_Before:_
+
+```groovy title="hello-channels.nf" linenums="11"
+process sayHello {
+
+    publishDir 'results', mode: 'copy'
+
+    input:
+        val greeting
+
+    output:
+        path 'output.txt'
+
+    script:
+    """
+    echo '$greeting' > output.txt
+    """
+}
+```
+
+_After:_
+
+```groovy title="hello-channels.nf" linenums="11"
+process sayHello {
+
+    publishDir 'results', mode: 'copy'
+
+    input:
+        val greeting
+
+    output:
+        path "${greeting}-output.txt"
+
+    script:
+    """
+    echo '$greeting' > '$greeting-output.txt'
+    """
+}
+```
+
+Make sure to replace `output.txt` in both the output definition and in the `script:` command block.
+
+!!! tip
+
+    In the output definition, you MUST use double quotes around the output filename expression (NOT single quotes), otherwise it will fail.
+
+This should produce a unique output file name every time the process is called.
+
+### 2.5. Run the workflow and look at the results directory
+
+Let's run it and check that it works.
+
+```bash
+nextflow run hello-channels.nf
+```
+
+Reverting back to the summary view, the output looks like this again:
+
+```console title="Output"
+ N E X T F L O W   ~  version 24.10.0
+
+ ┃ Launching `hello-channels.nf` [jovial_mccarthy] DSL2 - revision: 53f20aeb70
+
+executor >  local (3)
+[03/f007f2] sayHello (1)       [100%] 3 of 3 ✔
+```
+
+But more importantly, now we have three new files in addition to the one we already had in the `results` directory:
+
+```console title="Directory contents"
+results
+├── Bonjour-output.txt
+├── Hello-output.txt
+├── Holà-output.txt
+└── output.txt
+```
+
+Success! Now we can add as many greetings as we like without worrying about output files being overwritten.
+
+!!! note
+
+    In practice, naming files based on the input data itself is almost always impractical.
+    The better way to generate dynamic filenames is to use a samplesheet contain relevant metadata (such as unique sample IDs) and create a data structure called a 'map'.
+    We can then pass the map to the processes, which can be set up to select an appropriate identifier to generate the filenames.
+    You'll learn how to do that later in your Nextflow training.
```

Comment:
```suggestion
    In practice, naming files based on the input data itself is almost always impractical.
    The better way to generate dynamic filenames is to pass metatdata to a process along with the input files. We can derive that metadata from a sample sheet as we're reading the input files. 
    You'll learn how to do that later in your Nextflow training.
```

I think the current note is too complex for someone just getting used to the idea of channels.

Reply:
completely agree